src/m_Do/m_Do_main.cpp

```diff
@@ -1,72 +1,490 @@
-//
-// Generated by dtk
-// Translation Unit: m_Do_main.cpp
-//
+/**
+ * m_Do_main.cpp
+ * Main Initialization
+ */
 
 #include "m_Do/m_Do_main.h"
-#include "dolphin/types.h"
+#include "DynamicLink.h"
+#include "JSystem/JFramework/JFWSystem.h"
+#include "JSystem/JKernel/JKRAram.h"
+#include "JSystem/JKernel/JKRSolidHeap.h"
+#include "JSystem/JUtility/JUTAssert.h"
+#include "JSystem/JUtility/JUTConsole.h"
+#include "JSystem/JUtility/JUTGamePad.h"
+#include "JSystem/JUtility/JUTReport.h"
+#include "MSL_C/string.h"
+#include "c/c_dylink.h"
+#include "d/d_com_inf_game.h"
+#include "dolphin/os/OS.h"
+#include "f_ap/f_ap_game.h"
+#include "m_Do/m_Do_MemCard.h"
+#include "m_Do/m_Do_Reset.h"
+#include "m_Do/m_Do_audio.h"
+#include "m_Do/m_Do_controller_pad.h"
+#include "m_Do/m_Do_dvd_thread.h"
+#include "m_Do/m_Do_graphic.h"
+#include "m_Do/m_Do_machine.h"
+
+class JUTGamePad;
 
 /* 800056E0-80005748       .text version_check__Fv */
 void version_check() {
-    /* Nonmatching */
+    if (strcmp("05Sep2002", "05Sep2002") == 0 && strcmp("", "") == 0) {
+        return;
+    }
+
+    OSReport_Error("SDKのバージョンが一致しません。停止します\n");
+    while (true) {
+    }
 }
 
 /* 80005748-800057AC       .text CheckHeap1__9HeapCheckFv */
 void HeapCheck::CheckHeap1() {
-    /* Nonmatching */
+    s32 totalUsedSize = mHeap->getTotalUsedSize();
+    s32 freeSize = mHeap->getFreeSize();
+
+    if (mMaxTotalUsedSize < totalUsedSize)
+        mMaxTotalUsedSize = totalUsedSize;
+
+    if (mMaxTotalFreeSize > freeSize)
+        mMaxTotalFreeSize = freeSize;
 }
 
+// unused data?
+static u32 lit_2100[] = {0x3F800000, 0x3F800000, 0x3F800000, 0x3F800000, 0x3F800000, 0x3F800000};
```

Comment:
mind making these floats?

Reply:
didn't seem to work correctly when using floats